### PR TITLE
SWC-6149

### PR DESCRIPTION
**Pull request by nickgros**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -371,7 +371,8 @@
 								<copy file="${project.basedir}/node_modules/react/umd/react.production.min.js" tofile="src/main/webapp/generated/react.production.min.js" />
 								<copy file="${project.basedir}/node_modules/react-dom/umd/react-dom.production.min.js" tofile="src/main/webapp/generated/react-dom.production.min.js" />
 								<copy file="${project.basedir}/node_modules/react-transition-group/dist/react-transition-group.min.js" tofile="src/main/webapp/generated/react-transition-group.min.js" />
-								<copy file="${project.basedir}/node_modules/react-measure/dist/index.umd.js" tofile="src/main/webapp/generated/react-measure.js" />
+                                <copy file="${project.basedir}/node_modules/react-measure/dist/index.umd.js" tofile="src/main/webapp/generated/react-measure.js" />
+                                <copy file="${project.basedir}/node_modules/react-router-dom/umd/react-router-dom.js" tofile="src/main/webapp/generated/react-router-dom.js" />
```
Comment (nickgros) [inline pom.xml]:
```suggestion
								<copy file="${project.basedir}/node_modules/react-measure/dist/index.umd.js" tofile="src/main/webapp/generated/react-measure.js" />
								<copy file="${project.basedir}/node_modules/react-router-dom/umd/react-router-dom.js" tofile="src/main/webapp/generated/react-router-dom.js" />
```

File: src/main/java/org/sagebionetworks/web/client/mvp/AppPlaceHistoryMapper.java [hunk 2]
```diff
@@ -47,7 +48,47 @@
...
+		ACTDataAccessSubmissionsPlace.Tokenizer.class,
+		ACTDataAccessSubmissionDashboardPlace.Tokenizer.class,
+		ACTAccessApprovalsPlace.Tokenizer.class,
+		EmailInvitation.Tokenizer.class,
+		WikiDiff.Tokenizer.class,
+		PasswordResetSignedTokenPlace.Tokenizer.class,
+		DataAccessApprovalTokenPlace.Tokenizer.class,
+		PersonalAccessTokenPlace.Tokenizer.class,
+		DownloadCartPlace.Tokenizer.class,
+		DataAccessManagementPlace.Tokenizer.class
```
Comment (nickgros) [inline src/main/java/org/sagebionetworks/web/client/mvp/AppPlaceHistoryMapper.java:91]: Added DataAccessManagementPlace.Tokenizer

File: src/main/java/org/sagebionetworks/web/client/place/DataAccessManagementPlace.java [hunk 3]
```diff
@@ -1,13 +1,21 @@
 package org.sagebionetworks.web.client.place;
 
+import com.google.gwt.place.shared.Place;
 import com.google.gwt.place.shared.PlaceTokenizer;
 import com.google.gwt.place.shared.Prefix;
 
-public class DataAccessManagementPlace extends ParameterizedPlace {
```
Comment (nickgros) [inline src/main/java/org/sagebionetworks/web/client/place/DataAccessManagementPlace.java:6]: ParameterizedPlace was modifying the URL making it impossible to navigate directly to a subpage (e.g. #!DataAccessManagementPlace:default/Submissions)

File: src/main/webapp/Portal.html [hunk 4]
```diff
@@ -138,7 +138,8 @@
 			head.load("https://www.googletagmanager.com/gtag/js?id=G-CEKFPZDZX7");
 			head.load(reactPath);
 			head.load(reactDomPath);
-			head.load(reactQueryPath);
+            head.load('generated/react-router-dom.js')
+            head.load(reactQueryPath);
```
Comment (nickgros) [inline src/main/webapp/Portal.html]:
```suggestion
			head.load('generated/react-router-dom.js')
			head.load(reactQueryPath);
```

File: src/main/java/org/sagebionetworks/web/client/view/DataAccessManagementViewImpl.java [hunk 5]
```diff
@@ -30,13 +41,18 @@ public void setPresenter(DataAccessManagementPresenter presenter) {
         headerWidget.configure();
         headerWidget.refresh();
         Window.scrollTo(0,0);
+		render();
```
Comment (nickgros) [inline src/main/java/org/sagebionetworks/web/client/view/DataAccessManagementViewImpl.java]:
```suggestion
        render();
```

File: src/main/java/org/sagebionetworks/web/client/view/DataAccessManagementViewImpl.java [hunk 6]
```diff
@@ -30,13 +41,18 @@ public void setPresenter(DataAccessManagementPresenter presenter) {
...
     @Override
     public void render() {
-
-    }
+		ReactDOM.unmountComponentAtNode(reactComponent.getElement());
+		ReactDOM.render(
+				React.createElementWithSynapseContext(SRC.SynapseComponents.ReviewerDashboard, null, propsProvider.getJsInteropContextProps()),
+				reactComponent.getElement()
+		);
+	}
```
Comment (nickgros) [inline src/main/java/org/sagebionetworks/web/client/view/DataAccessManagementViewImpl.java]:
```suggestion
        ReactDOM.unmountComponentAtNode(reactComponent.getElement());
        ReactDOM.render(
            React.createElementWithSynapseContext(SRC.SynapseComponents.ReviewerDashboard, null, propsProvider.getJsInteropContextProps()),
            reactComponent.getElement()
        );
    }
```

File: src/main/java/org/sagebionetworks/web/client/view/DataAccessManagementViewImpl.java [hunk 7]
```diff
@@ -13,14 +19,19 @@ public interface DataAccessManagementViewImplUiBinder extends UiBinder<Widget, D
...
+	ReactComponentDiv reactComponent;
+
     Widget widget;
 
     @Inject
-    public DataAccessManagementViewImpl(DataAccessManagementViewImplUiBinder binder, Header headerWidget) {
+    public DataAccessManagementViewImpl(DataAccessManagementViewImplUiBinder binder, Header headerWidget, SynapseContextPropsProvider propsProvider) {
         widget = binder.createAndBindUi(this);
         this.headerWidget = headerWidget;
+		this.propsProvider = propsProvider;
```
Comment (nickgros) [inline src/main/java/org/sagebionetworks/web/client/view/DataAccessManagementViewImpl.java]:
```suggestion
        this.propsProvider = propsProvider;
```

File: src/main/java/org/sagebionetworks/web/client/view/DataAccessManagementViewImpl.java [hunk 8]
```diff
@@ -13,14 +19,19 @@ public interface DataAccessManagementViewImplUiBinder extends UiBinder<Widget, D
     }
 
     private DataAccessManagementPresenter presenter;
+	private SynapseContextPropsProvider propsProvider;
     private Header headerWidget;
 
+	@UiField
+	ReactComponentDiv reactComponent;
```
Comment (nickgros) [inline src/main/java/org/sagebionetworks/web/client/view/DataAccessManagementViewImpl.java]:
```suggestion
    private SynapseContextPropsProvider propsProvider;
    private Header headerWidget;

    @UiField
    ReactComponentDiv reactComponent;
```

File: src/main/java/org/sagebionetworks/web/client/place/DataAccessManagementPlace.java [hunk 9]
```diff
@@ -1,13 +1,21 @@
...
     public DataAccessManagementPlace(String token) {
-        super(token);
-    }
+		this.token = token;
+
+	}
+
+	public String toToken() {
+		return token;
+	}
```
Comment (nickgros) [inline src/main/java/org/sagebionetworks/web/client/place/DataAccessManagementPlace.java]:
```suggestion
        this.token = token;
    }

    public String toToken() {
        return token;
    }
```

File: src/main/java/org/sagebionetworks/web/client/place/DataAccessManagementPlace.java [hunk 10]
```diff
@@ -1,13 +1,21 @@
 package org.sagebionetworks.web.client.place;
 
+import com.google.gwt.place.shared.Place;
 import com.google.gwt.place.shared.PlaceTokenizer;
 import com.google.gwt.place.shared.Prefix;
 
-public class DataAccessManagementPlace extends ParameterizedPlace {
+public class DataAccessManagementPlace extends Place {
+
+	String token;
```
Comment (nickgros) [inline src/main/java/org/sagebionetworks/web/client/place/DataAccessManagementPlace.java]:
```suggestion
    String token;
```

File: pom.xml [hunk 11]
```diff
@@ -373,6 +373,7 @@
 								<copy file="${project.basedir}/node_modules/react-router-dom/umd/react-router-dom.min.js" tofile="src/main/webapp/generated/react-router-dom.min.js" />
 								<copy file="${project.basedir}/node_modules/react-transition-group/dist/react-transition-group.min.js" tofile="src/main/webapp/generated/react-transition-group.min.js" />
 								<copy file="${project.basedir}/node_modules/react-measure/dist/index.umd.js" tofile="src/main/webapp/generated/react-measure.js" />
+								<copy file="${project.basedir}/node_modules/react-router-dom/umd/react-router-dom.js" tofile="src/main/webapp/generated/react-router-dom.js" />
```
Comment (nickgros) [inline pom.xml]:
```suggestion
```

File: src/main/webapp/Portal.html [hunk 12]
```diff
@@ -138,6 +138,7 @@
 			head.load("https://www.googletagmanager.com/gtag/js?id=G-CEKFPZDZX7");
 			head.load(reactPath);
 			head.load(reactDomPath);
+			head.load('generated/react-router-dom.js')
```
Comment (nickgros) [inline src/main/webapp/Portal.html]:
```suggestion
```

**Review by jay-hodgson — APPROVED** [review]

File: src/main/java/org/sagebionetworks/web/client/place/DataAccessManagementPlace.java (hunk 3 above)
Comment (jay-hodgson) [reply src/main/java/org/sagebionetworks/web/client/place/DataAccessManagementPlace.java:6]: Perhaps unrelated, but why could it not just be #!DataAccessManagementPlace:Submissions ?

**Pull request merged**